### PR TITLE
Added INISerializer, which help benedict parsing and writing to .ini

### PR DESCRIPTION
**Pull request by antran22**
Parse and write `.ini` file with ConfigParser.
When writing object to `.ini` file, I've process the data by the following rule:
For all the key-value pair in the dict:
- If the value is a not a dict, put them into the default namespace, specified by the `default_namespace` argument
- If the value is a dict, put the dict as a new section
Then run through the sections, if a value is an object, turn it into string.

### CODE REVIEW

File: benedict/dicts/io/io_dict.py [hunk 1]
```diff
@@ -183,6 +185,16 @@ def to_json(self, **kwargs):
         """
         return self._encode(self, 'json', **kwargs)
 
+    def to_ini(self, **kwargs):
+        """
```
Comment (fabiocaccamo) [inline benedict/dicts/io/io_dict.py:189]: Wrong comment.

File: benedict/dicts/io/io_dict.py [hunk 2]
```diff
@@ -39,15 +39,17 @@ def _decode(s, format, **kwargs):
                 return data
             elif type_util.is_list(data):
                 # force list to dict
-                return {'values': data}
+                return {
```
Comment (fabiocaccamo) [inline benedict/dicts/io/io_dict.py:42]: These changes are not related to this PR.

File: benedict/serializers/__init__.py [hunk 3]
```diff
@@ -37,6 +39,7 @@
     'yaml': _YAML_SERIALIZER,
     'yml': _YAML_SERIALIZER,
     'xml': _XML_SERIALIZER,
+    'ini': _INI_SERIALIZER,
```
Comment (fabiocaccamo) [inline benedict/serializers/__init__.py:42]: These entries are sorted alphabetically.

**Review by fabiocaccamo — CHANGES_REQUESTED** [review]

**Comment by fabiocaccamo** [comment]
@antran22 I appreciate the effort, but I can't merge a pull request that doesn't have tests.
A new format should be tested in these contexts: 
- using dict: https://github.com/fabiocaccamo/python-benedict/tree/master/tests/dicts/io
- only serializer: https://github.com/fabiocaccamo/python-benedict/tree/master/tests/serializers

**Comment by antran22** [comment]
Yes, I will be working on this

**Comment by fabiocaccamo** [comment]
You can use existing tests as reference, and keep in mind that doing this must not raise an exception:  

```python
d = { ... }
b = benedict(d)
assert b == benedict.from_ini(b.to_ini())
```

File: benedict/dicts/io/io_dict.py (hunk 2 above)
Comment (antran22) [reply benedict/dicts/io/io_dict.py:42]: This is probably because of my code formatter. I will be reverting it back.

File: benedict/serializers/__init__.py (hunk 3 above)
Comment (antran22) [reply benedict/serializers/__init__.py:42]: Oh alright, this can be fixed

File: benedict/serializers/ini.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 from __future__ import absolute_import
 
 import io
-from typing import Dict
+from copy import deepcopy
```
Comment (fabiocaccamo) [inline benedict/serializers/ini.py:6]: Instead of using `copy.deepcopy`, you could use `benedict.core.clone` method, which is a `deepcopy` wrapper, in this way there will be a single entry point for all `deepcopy` calls.

File: tests/dicts/io/test_io_dict_ini.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
+
+from benedict.dicts.io import IODict
+
+from .test_io_dict import io_dict_test_case
+
+TARGET_DICT = {
```
Comment (fabiocaccamo) [inline tests/dicts/io/test_io_dict_ini.py:7]: The test dictionary should cover most use cases as possible:
-  Could you add at least one entry for each data-type?   
-  Even doing tests against both nested dicts and lists is recommended.
-  String entries containing special characters (eg. àèéìòùç) must be tested.

File: tests/serializers/test_ini_serializer.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+
+from benedict.serializers import INISerializer, PickleSerializer
```
Comment (fabiocaccamo) [inline tests/serializers/test_ini_serializer.py:3]: Why `PickleSerializer` here?

File: tests/serializers/test_ini_serializer.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+class ini_serializer_test_case(unittest.TestCase):
+    def test_decode_pickle(self):
+        d = INISerializer().decode(INI_STR)
+        self.assertEqual(d, TARGET_DICT)
+
+    def test_encode_pickle(self):
+        s = INISerializer().encode(TARGET_DICT)
+        self.assertEqual(s, INI_STR)
+
+    def test_encode_decode_pickle(self):
```
Comment (fabiocaccamo) [inline tests/serializers/test_ini_serializer.py:42]: Wrong test declaration and implementation.

**Review by fabiocaccamo — CHANGES_REQUESTED** [review]

File: tests/dicts/io/test_io_dict_ini.py (hunk 5 above)
Comment (antran22) [reply tests/dicts/io/test_io_dict_ini.py:7]: The problem with .ini files is that they can only support a two level nested object, so the real world use case is also really limited.

Comment (antran22) [reply tests/dicts/io/test_io_dict_ini.py:7]: As for the datatypes, I have put str, float, int and boolean which are the types supported by ini files. Any other object will be converted to string.

Comment (fabiocaccamo) [reply tests/dicts/io/test_io_dict_ini.py:7]: Ok, so there should be a validation that will raise an exception in case the dict that should be converted has more than 2 levels.

**Review by fabiocaccamo — APPROVED** [review]

**Comment by fabiocaccamo** [comment]
@antran22 your PR doesn't pass the travis CI build, you should debug travis build error and fix them.

File: benedict/serializers/ini.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import
+
+import io
+import configparser
+
+from benedict.core import clone
+from benedict.serializers.abstract import AbstractSerializer
+
```
Comment (fabiocaccamo) [inline benedict/serializers/ini.py:10]: Imports should be sorted alphabetically.

File: tests/serializers/test_ini_serializer.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+[section_b]
+c = 2.5
+d = True
+e = False
+
+"""
+
+
+class ini_serializer_test_case(unittest.TestCase):
+    def test_decode_pickle(self):
```
Comment (fabiocaccamo) [inline tests/serializers/test_ini_serializer.py:32]: Wrong method name.

File: tests/serializers/test_ini_serializer.py [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+        d = INISerializer().decode(INI_STR)
+        self.assertEqual(d, TARGET_DICT)
+
+    def test_encode_pickle(self):
```
Comment (fabiocaccamo) [inline tests/serializers/test_ini_serializer.py:36]: Wrong method name.

**Review by fabiocaccamo — CHANGES_REQUESTED** [review]
There are some minor things that should be fixed.
Once the changes will be done, please check the results of the CI and if it fails you should check and fix the errors.
